proposed/http-message.md

```diff
@@ -146,6 +146,27 @@ present in the composed `UriInterface`, the value from the URI should be used.
 If a Host header is explicitly provided to the request instance, that value will
 be preferred.
 
+`RequestInterface::withUri()` also interacts with the Host header. By default,
+this method replaces the returned request's Host header with a Host header
+matching the host component of the passed `UriInterface`.
+
+You can opt-in to preserving the original state of the Host header by passing
+`true` for the second (`$preserveHost`) argument. When this argument is set to
+`true`, the returned request will not update the Host header of the returned
+message -- unless the message contains no Host header.
+
+This table illustrates what `getHeaderLine('Host')` will return for a request
+returned by `withUri()` with the `$preserveHost` argument set to `true` for
+various initial requests and URIs.
+
+Request Host header | Request host component | URI host component | Result
+--------------------|------------------------|--------------------|--------
+''                  | ''                     | ''                 | ''
+''                  | foo.com                | ''                 | foo.com
```

Comment:
@Tobion — a follow-up: `RequestInterface` notes in its `getHeader*()` methods:

> if the Host header is requested, but has not been previously set, the method MUST attempt to pull the host component of the composed URI, if present.

This is why you're seeing the result of `foo.com`: the composed URI has a host component.

That said: @pjdietz, this example is wrong. If you set a new URI without a host component, and do not have a `Host` header currently, the behavior should be to have an empty `Host` header.